gslib/commands/iam.py

```diff
@@ -363,8 +374,153 @@ def get_gcloud_storage_args(self):
       # Gsutil takes policy file then URLs, and gcloud storage does the reverse.
       self.args = url_strings + self.args[:1]
 
+    elif self.sub_command == 'ch':
+      if shim_util.HIDDEN_SHIM_MODE(
+          config.get('GSUtil', 'hidden_shim_mode',
+                     'none')) is shim_util.HIDDEN_SHIM_MODE.DRY_RUN:
+        self.logger.warning(
+            'The shim maps iam ch commands to several gcloud storage commands,'
+            ' which cannot be determined without running gcloud storage.')
+      return []  # Translation occurs in self.run_gcloud_storage.
+
     return super().get_gcloud_storage_args(gcloud_storage_map)
 
+  def _RaiseIfInvalidUrl(self, url):
+    if url.IsFileUrl():
+      error_msg = 'Invalid Cloud URL "%s".' % url.object_name
+      if set(url.object_name).issubset(set('-Rrf')):
```

Comment:
Can you just do `url.object_name in ('-R', '-r', '-f')`? What about `-d`?

Reply:
This is the existing gsutil check, and I'd prefer to keep it the same for consistency/compatibility as it concerns `-d`. Additionally, `url.object_name in ('-R', '-r', '-f')` does have different behavior than what's here, since the existing check handles combined flags like `-rf`.